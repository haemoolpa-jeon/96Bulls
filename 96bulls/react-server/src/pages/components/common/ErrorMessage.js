/* Error Message when error occurs */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './ErrorMessage.module.scss';

class ErrorMessage extends Component {
  render() {
    const { children } = this.props;
    return (
      <div className={styles.message}>
        <h3 className={styles.title}>ERROR</h3>
        {children}
      </div>
    );
  }
}

ErrorMessage.propTypes = {
  children: PropTypes.node.isRequired,
};

export default ErrorMessage;
