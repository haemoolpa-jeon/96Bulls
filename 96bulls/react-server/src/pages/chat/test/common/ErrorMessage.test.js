import React from 'react';
import ErrorMessage from '../../components/common/ErrorMessage';
import {shallow} from 'enzyme/build';

describe('ErrorMessage', () => {
  let component = null;
  it('초기 렌더링이 문제없이 되야함', () => {
    component = shallow(<ErrorMessage>
      <p>error message</p>
    </ErrorMessage>);
  });

  it('초기 렌더링 스냅샷 일치함', () => {
    expect(component).toMatchSnapshot();
  });
});
