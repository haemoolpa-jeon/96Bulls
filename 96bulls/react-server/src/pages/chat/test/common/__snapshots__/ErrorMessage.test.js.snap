// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage 초기 렌더링 스냅샷 일치함 1`] = `
<div
  className="message"
>
  <h3
    className="title"
  >
    (ㅇ.ㅇ)!!
  </h3>
  <p>
    error message
  </p>
</div>
`;
